backend/app/services/query_service.py

```
@@ -0,0 +1,471 @@
+                sources=cached_response.get("sources") or [],
+            )
+            # Append this Q&A to chat history so conversation stays consistent
+            cache_key = cache_key_chat_history(user_id, session_id)
+            raw_history = await cache_get(cache_key)
+            history_list = list(raw_history) if isinstance(raw_history, list) else []
+            history_list.append({"role": "human", "content": question})
+            history_list.append({"role": "ai", "content": response.answer})
+            await cache_set(cache_key, history_list[-(history_window * 2) :], ttl_seconds=ttl)
+            return response
+        except Exception:
+            pass
```

Ruff (0.15.5)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)